java/vector/src/test/java/org/apache/arrow/vector/validate/TestValidateVectorFull.java

```diff
@@ -0,0 +1,158 @@
+      offsetBuf.setInt(0, 100);
+      offsetBuf.setInt(4, 50);
+
+      ValidateUtility.ValidateException e = assertThrows(ValidateUtility.ValidateException.class,
+          () -> validateFull(vector));
+      assertTrue(e.getMessage().contains("The values in the large offset buffer are decreasing"));
+    }
+  }
+
+  @Test
+  public void testListVector() {
+    try (final ListVector vector = ListVector.empty("v", allocator)) {
+      validateFull(vector);
+      setVector(vector, Arrays.asList(1, 2, 3), Arrays.asList(4, 5), Arrays.asList(6, 7, 8, 9));
+      validateFull(vector);
+
+      ArrowBuf offsetBuf = vector.getOffsetBuffer();
+      offsetBuf.setInt(0, 100);
+      offsetBuf.setInt(8, 50);
+
+      ValidateUtility.ValidateException e = assertThrows(ValidateUtility.ValidateException.class,
+          () -> validateFull(vector));
+      assertTrue(e.getMessage().contains("The values in the offset buffer are decreasing"));
+    }
+  }
+
+  @Test
+  public void testLargeListVector() {
+    try (final LargeListVector vector = LargeListVector.empty("v", allocator)) {
+      validateFull(vector);
+      setVector(vector, Arrays.asList(1, 2, 3), Arrays.asList(4, 5), Arrays.asList(6, 7, 8, 9));
+      validateFull(vector);
+
+      ArrowBuf offsetBuf = vector.getOffsetBuffer();
+      offsetBuf.setLong(0, 100);
+      offsetBuf.setLong(16, 50);
+
+      ValidateUtility.ValidateException e = assertThrows(ValidateUtility.ValidateException.class,
+          () -> validateFull(vector));
+      assertTrue(e.getMessage().contains("The values in the large offset buffer are decreasing"));
+    }
+  }
+
+  @Test
+  public void testStructVectorRangeEquals() {
+    try (final StructVector vector = StructVector.empty("struct", allocator)) {
+      IntVector intVector =
+          vector.addOrGet("f0", FieldType.nullable(new ArrowType.Int(32, true)), IntVector.class);
+      VarCharVector strVector =
+          vector.addOrGet("f1", FieldType.nullable(new ArrowType.Utf8()), VarCharVector.class);
+
+      validateFull(vector);
+      validateFull(intVector);
+      validateFull(strVector);
+
+      ValueVectorDataPopulator.setVector(intVector, 1, 2, 3, 4, 5);
+      ValueVectorDataPopulator.setVector(strVector, "a", "b", "c", "d", "e");
+      vector.setValueCount(5);
+
+      validateFull(vector);
+      validateFull(intVector);
+      validateFull(strVector);
+
+      ArrowBuf offsetBuf = strVector.getOffsetBuffer();
+      offsetBuf.setInt(0, 100);
+      offsetBuf.setInt(8, 50);
+
+      ValidateUtility.ValidateException e = assertThrows(ValidateUtility.ValidateException.class,
+          () -> validateFull(strVector));
+      assertTrue(e.getMessage().contains("The values in the offset buffer are decreasing"));
+
+      e = assertThrows(ValidateUtility.ValidateException.class,
+          () -> validateFull(vector));
+      assertTrue(e.getMessage().contains("The values in the offset buffer are decreasing"));
+    }
+  }
+}
```

Comment:
Should we test Unions?

Reply:
We have added tests for union and dense union vectors.